torch/_dynamo/eval_frame.py

```diff
@@ -331,6 +331,16 @@ def _fn(*args, **kwargs):
             dynamic_ctx.__enter__()
             try:
                 return fn(*args, **kwargs)
+            except Exception as e:
```

Comment:
This seems like the wrong place to put this. The main problem is that calling fn doesn't actually guarantee that you compiled. If you hit a skip frame, we may have already executed a bunch of eager code along the way. You want to catch the exception *in the frame callback*, and call into eager inside the C code, IMO.